Sources/UtiliKit/ExternalMapping/ExternalMappingURLBuilder.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  ExternalMappingURLBuilder.swift
+//  UtiliKit-iOS
+//
+//  Created by Nathan Chiu on 9/24/21.
+//  Copyright © 2021 Bottle Rocket Studios. All rights reserved.
+//
+
+import Foundation
+
+public struct ExternalMappingURLBuilder {
+
+    // MARK: Properties
+
+    private let apps: [MapApp]
+
+    // MARK: Initializers
+
+    /// Create an instance with the external mapping apps
+    /// - Parameter apps: The apps the builder will build links to. Optional, defaults to all supported apps.
+    public init(apps: [MapApp] = MapApp.allCases) {
+        self.apps = apps
+    }
+
+    // MARK: Public
+
+    /// Creates links to display a specific location on the map.
+    /// - Parameters:
+    ///   - coordinate: The location on the map to display.
+    ///   - zoomPercent: The zoom level specified on a `0.0...1.0` scale. `1.0` represents fully zoomed in on the location. Optional, defaults to `nil`.
+    ///   - style: The map style to display. Optional, defaults to `nil`.
+    /// - Returns: URLs to display the location in each of the supported map apps.
+    public func displayLocation(at coordinate: MappingCoordinate, zoomPercent: Float?, style: MapStyle?) -> [MapApp: URL] {
+        var returnDictionary = [MapApp: URL]()
+        apps.forEach {
```

Comment:
I believe this could be done with a reduce(). This isn't complete, but it should be 95%.

```suggestion
apps.reduce(into: [:]) { partialResult, element in
  partialResult[$0] = $0.interface.displayLocation(at: coordinate, zoomPercent: ZoomPercent, style: style)
}```